File: 2180/2180.c
#include <stdio.h>

int check(int num) {
    int check = 0;
    while (num) {
        check += num % 10;
        num /= 10;
    }
    return !(check % 2);
}

int countEven(int num){
    int ret = 0;
    for (int i = 1; i <= num; ++i) {
        if (check(i)) {
            ret++;
        }
    }
    return ret;
}

int main() {
    for (int i = 1; i <= 1000; ++i) {
        printf("%d ", countEven(i));
    }
    return 0;
}

File: 2180/2180.rs
struct Solution {}

impl Solution {
    pub fn count_even(num: i32) -> i32 {
        let mut tmp: i32 = num;
        let mut check: i32 = 0;
        while (tmp > 0) {
            check += tmp % 10;
            tmp /= 10;
        }
        if check & 1 == 1 {
            return (num-1) / 2; // go back to find the biggest even sumation
        }
        num / 2
    }
}
